pkg/operator/baremetal_pod.go

```diff
@@ -46,14 +56,56 @@ func setMariadbPassword() corev1.EnvVar {
 		ValueFrom: &corev1.EnvVarSource{
 			SecretKeyRef: &corev1.SecretKeySelector{
 				LocalObjectReference: corev1.LocalObjectReference{
-					Name: "mariadb-password",
+					Name: baremetalSecretName,
 				},
-				Key: "password",
+				Key: baremetalSecretKey,
 			},
 		},
 	}
 }
 
+func generateRandomPassword() string {
```

Comment:
I wonder if there is a library we could use for this?  Generally I've always been taught to use trusted libs for crypto/random generators etc.

Reply:
I am using math/rand library to generate the random string using the unicode characters in "chars". rand is seeded with the current time so every time we try to generate a random password, the seed is different.